### PR TITLE
Http Class Name Cleanup

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR standardizes the naming conventions used by our HttpClient implementations, most importantly it standardizes the SPI implementation name which has recently been moved into the public API.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@JonathanGiles, @srnagar could you review the name changes as this should be merged before shipping azure-core again.

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncClientProvider.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 /**
  * An {@link HttpClientProvider} that provides an implementation of HttpClient based on OkHttp.
  */
-public final class OkHttpClientProvider implements HttpClientProvider {
+public final class OkHttpAsyncClientProvider implements HttpClientProvider {
```
Comment (srnagar) [inline sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncClientProvider.java:13]: This is a breaking change. The package and the class both are public.

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 /**
  * An {@link HttpClientProvider} that provides an implementation of HttpClient based on Netty.
  */
-public final class ReactorNettyClientProvider implements HttpClientProvider {
+public final class NettyAsyncHttpClientProvider implements HttpClientProvider {
```
Comment (srnagar) [inline sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java:13]: This is a breaking change. The package and the class both are public.

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncClientProvider.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncClientProvider.java:13]: These were moved into public API earlier this month and haven't been released yet

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java:13]: These were moved into public API earlier this month and haven't been released yet

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/Utility.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import io.netty.buffer.ByteBuf;
+import reactor.netty.Connection;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Helper class containing utility methods.
+ */
+public final class Utility {
```
Comment (JonathanGiles) [inline sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/Utility.java:14]: Not a huge fan of how many Utility classes we have - can this be made into, e.g. NettyUtility or similar

**Review by JonathanGiles — APPROVED** [review]

**Comment by alzimmermsft** [comment]
The CI checks from the manually triggered run passed

https://dev.azure.com/azure-sdk/public/_build/results?buildId=715867&view=results

**Pull request merged**